### PR TITLE
@feature/api implementation

### PR DESCRIPTION
**Pull request by TingChiehLin**
## Summary

 - CSS and naming
-  Data manipulation
- Object pooling?? (read fast with key and value)

## Struggles

- When I fetch data to specific Australia as key and then need to get this key to fetch the other endpoint for new data rending

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,68 @@
...
 
-You can start editing the page by modifying `app/page.tsx`. The page auto-updates as you edit the file.
+## Tech I used
 
-[API routes](https://nextjs.org/docs/api-routes/introduction) can be accessed on [http://localhost:3000/api/hello](http://localhost:3000/api/hello). This endpoint can be edited in `pages/api/hello.ts`.
+- React
+- Next.js 13
+- CSS3 Module
+- React Query
+- Test Library()
```
Comment (sandrina-p) [inline README.md]: fix: You mean Testing Library

File: components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+interface ButtonTypeProp {
+  text: string;
+  onClickEvent: () => void;
+}
+
+const Button: FC<ButtonTypeProp> = ({ text, onClickEvent }) => {
+  return (
+    <button className={styles.button_container} onClick={onClickEvent}>
+      {text}
```
Comment (sandrina-p) [inline components/Button/Button.tsx]: suggestion: Usually text in button is more common to pass as children, so you use it like `<Button>start</Button>`

File: components/Button/button.module.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  width: 100%;
+  max-width: 10rem;
+  padding: 1.2rem 1.5rem;
+  border-radius: 10px;
+  outline: none;
+  border: 0px;
+  cursor: pointer;
+}
+
+.button_container:active {
```
Comment (sandrina-p) [inline components/Button/button.module.css]: q: what do you mean by `active` here?  Maybe what you meant is `:focus`?

File: components/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { FC, useState } from "react";
+import styles from "./checkbox.module.css";
+
+interface CheckboxPropType {}
+
+const Checkbox: FC<CheckboxPropType> = () => {
```
Comment (sandrina-p) [inline components/Checkbox/Checkbox.tsx]: todo: It's missing some things in this component:

- An accessible label with `aria-label` ([reference](https://css-tricks.com/html-inputs-and-labels-a-love-story/))
- To connect the `id` to the label. `htmlFor` cannot be empty

File: public/desktop_demo.png
Comment (sandrina-p) [inline public/desktop_demo.png:1]: Please do not use these screenshots in the repo. Instead, delete them and take a screenshot of how the app really looks.

File: app/layout.tsx [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
-  children: React.ReactNode
+  children: React.ReactNode;
 }) {
   return (
     <html lang="en">
-      <body>{children}</body>
+      <body>
+        {children}
+        <footer>
+          <span>Developer by Ting Chieh Lin</span>
```
Comment (sandrina-p) [inline app/layout.tsx]: grammar: here the correct english is "Developer by *",

suggestion: You could add the URL to your github :)

File: components/Button/button.module.css (hunk 3 above)
Comment (TingChiehLin) [reply components/Button/button.module.css]: active mean when the user clicks the button and has status?
when should I use focus?

File: components/Checkbox/Checkbox.tsx (hunk 4 above)
Comment (TingChiehLin) [reply components/Checkbox/Checkbox.tsx]: not use it anymore

**Comment by TingChiehLin** [comment]
Update API and filter

File: README.md [hunk 6]
```diff
@@ -1,38 +1,68 @@
...
-The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+To learn more from this project, take a look at the following resources:
+
+- [Border Property](https://stackoverflow.com/questions/628301/the-border-radius-property-and-border-collapsecollapse-dont-mix-how-can-i-use) - To resolve border issue
+- [React Query Course](https://www.youtube.com/watch?v=NQULKpW6hK4) - data-fetching library with fetching, caching, synchronising, and updating the server state in my react application
+
+## Deploy on Vercel
 
-Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+You can see my project on Vercel [Demo Link](https://world-holidays.vercel.app)
```
Comment (sandrina-p) [inline README.md]: suggestion: As I mentioned yesterday, I'd put the Demo link at the top of the readme to be more easily seen :)

File: public/desktop_demo.png
Comment (sandrina-p) [inline public/desktop_demo.png:1]: I'd appreciate it if instead of using my Figma mockups, you take real screenshots of what you implemented.

**Review by sandrina-p — COMMENTED** [review]
I sent you the entire review on Discord, go check it out! :)

File: README.md [hunk 7]
```diff
@@ -62,6 +62,7 @@ To learn more from this project, take a look at the following resources:
 
 - [Border Property](https://stackoverflow.com/questions/628301/the-border-radius-property-and-border-collapsecollapse-dont-mix-how-can-i-use) - To resolve border issue
 - [React Query Course](https://www.youtube.com/watch?v=NQULKpW6hK4) - data-fetching library with fetching, caching, synchronising, and updating the server state in my react application
+- [](https://www.w3.org/WAI/tutorials/images/decorative/) - Understung Decorative Images for img tag
```
Comment (sandrina-p) [inline README.md]: You can a typo here "Understung" -> "Understanding"

Comment (sandrina-p) [reply README.md]: And it's missing the text for this link, as it's empty.

**Pull request merged**